parity-clib/examples/cpp/main.cpp

```diff
@@ -14,172 +14,134 @@
+      std::regex is_subscription(
+          "\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-"
+          "F0-9]{16}\",\"id\":1\\}");
+      if (std::regex_match(response.begin(), response.end(), results, is_subscription)) {
+        counter -= 1;
+      }
+    } else {
+      abort();
+    }
+  }
 };
 
 // list of rpc queries
-const std::vector<std::string> rpc_queries {
-	"{\"method\":\"parity_versionInfo\",\"params\":[],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"0x444172bef57ad978655171a8af2cfd89baa02a97fcb773067aef7794d6913fff\"],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_estimateGas\",\"params\":[{\"from\":\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"}],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_getBalance\",\"params\":[\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
+const std::vector<std::string> rpc_queries{
+    "{\"method\":\"parity_versionInfo\",\"params\":[],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"0x444172bef57ad978655171a8af2cfd89baa02a97fcb773067aef7794d6913fff\"],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_estimateGas\",\"params\":[{\"from\":\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"}],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_getBalance\",\"params\":[\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
 };
 
 // list of subscriptions
-const std::vector<std::string> ws_subscriptions {
-	"{\"method\":\"parity_subscribe\",\"params\":[\"eth_getBalance\",[\"0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826\",\"latest\"]],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"parity_subscribe\",\"params\":[\"parity_netPeers\"],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_subscribe\",\"params\":[\"newHeads\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
+const std::vector<std::string> ws_subscriptions{
+    "{\"method\":\"parity_subscribe\",\"params\":[\"eth_getBalance\",[\"0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826\",\"latest\"]],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"parity_subscribe\",\"params\":[\"parity_netPeers\"],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_subscribe\",\"params\":[\"newHeads\"],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
 };
 
 // callback that gets invoked upon an event
-void callback(void* user_data, const char* response, size_t _len) {
-	Callback* cb = static_cast<Callback*>(user_data);
-	if (cb->type == CALLBACK_RPC) {
-		cb->counter -= 1;
-	} else if (cb->type == CALLBACK_WS) {
-		std::regex is_subscription ("\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-F0-9]{16}\",\"id\":1\\}");
-		if (std::regex_match(response, is_subscription) == true) {
-			cb->counter -= 1;
-		}
-	}
+void callback(std::string_view buf) { (void)buf; }
+} // namespace
+
+int main(int _argc, char **_argv) {
+  (void)_argc, (void)_argv;
+  using parity::ethereum::ParityEthereum;
+  // run full-client
+  if (0) {
```

Comment:
? :upside_down_face: 

also please prefer `false` or `true` in `bool` expressions.

Reply:
That was left from when I was trying to debug the segmentation faults, and didn’t want to sync the Kovan testnet.  Very restrictive firewall rules on my system would have stopped it anyway, but I still didn’t want it to try.  Now fixed.